File: bench/Reference/IntArrayMap.c
/* C reference: map (*2) then sum 10K elements */
#include <stdio.h>
#include <stdint.h>

int main(void) {
    int64_t arr[10000], doubled[10000];
    for (int i = 0; i < 10000; i++) arr[i] = i + 1;
    for (int i = 0; i < 10000; i++) doubled[i] = arr[i] * 2;
    int64_t sum = 0;
    for (int i = 0; i < 10000; i++) sum += doubled[i];
    printf("%ld\n", sum);
    return 0;
}

File: bench/Reference/int_array_map.js
const a = new Int32Array(10000);
for (let i = 0; i < 10000; i++) a[i] = i + 1;
const doubled = new Int32Array(10000);
for (let i = 0; i < 10000; i++) doubled[i] = a[i] * 2;
let s = 0;
for (let i = 0; i < doubled.length; i++) s += doubled[i];
console.log(s);
